pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -923,6 +935,17 @@ var (
 		},
 	}
 
+	PortRuleL7 = apiextensionsv1beta1.JSONSchemaProps{
+		Description: "PortRuleL7 is a map of {key,value} pairs which must be present in the " +
```

Comment:
Non-blocking: I think the description is a bit too specific. We may use this for things other than policy. I would reword this to a generic statement that says that the map is passed to the parser referenced in l7proto and it is up to the parser to define what to do with the map data.

Reply:
Changed, but kept the note that "If omitted or empty, all requests are allowed", as that is part of the matching logic that is independent of the L7. This allows cilium-agent to create wildcard policies when needed without understanding the specifics of a L7 protocol and is also the way HTTP policies work.

Currently the L7 implementation has to call the matching logic, though, so a parser could be written to violate this rule.